src/main/java/duke/logic/parsers/Parser.java

```diff
@@ -0,0 +1,139 @@
+package duke.logic.parsers;
+
+import duke.logic.commands.AddCommand;
+import duke.logic.commands.Command;
+import duke.logic.commands.DeleteCommand;
+import duke.logic.commands.ExitCommand;
+import duke.logic.commands.FindCommand;
+import duke.logic.commands.FindPathCommand;
+import duke.logic.commands.FreeTimeCommand;
+import duke.logic.commands.GetBusRouteCommand;
+import duke.logic.commands.GetBusStopCommand;
+import duke.logic.commands.HelpCommand;
+import duke.logic.commands.ListCommand;
+import duke.logic.commands.LocationSearchCommand;
+import duke.logic.commands.MarkDoneCommand;
+import duke.logic.commands.PromptCommand;
+import duke.logic.commands.RecommendationsCommand;
+import duke.logic.commands.ReminderCommand;
+import duke.logic.commands.RescheduleCommand;
+import duke.logic.commands.StaticMapCommand;
+import duke.logic.commands.ViewScheduleCommand;
+import duke.commons.MessagesPrompt;
+import duke.commons.exceptions.DukeException;
+import duke.commons.exceptions.DukeUnknownCommandException;
+
+
+/**
+ * Parser for commands entered by the user. It reads from standard input and
+ * returns Command objects.
+ */
+public class Parser {
+    /**
+     * Parses the userInput and return a Command object.
+     *
+     * @param input Input created by the ConversationManager object or user input.
+     * @return The corresponding Command object.
+     * @throws DukeException If userInput is undefined.
+     */
+    public static Command parseComplexCommand(String input) throws DukeException {
+        String commandWord = getCommandWord(input);
+        switch (commandWord) {
+        case "todo":
```

Comment:
did I see something above which looks similar to this part of the code?

Reply:
We are unable to come up with an elegant way to merge these overlaps. As ```todo``` here can trigger both the multi-line and single-line command which depends on the context of the user input. 

For example, single-line command is triggered when user inputs ```todo something```, whereas ```todo``` triggers the multi-line command.

However, in multi-line command, the command itself is not executed until the description of todo is provided by the user in this scenario, thus, the parser should not return a  AddCommand object, but a PromptCommand object to prompt for more information. But if it were to return a PromptCommand object, it means that the parser must have analyse the context of the user input to even determine that a PromptCommand should be return. 

To analyse the context, the Parser would be doing what the Logic class is responsible for and violating SRP and SOC.